### PR TITLE
Feature/45 microsoft api standard

### PR DESCRIPTION
**Pull request by RamonG92**

### CODE REVIEW

**Comment by CalvinHuynh** [comment]
Please extend the new promises to all the other controllers

File: src/helpers/response.helper.ts [hunk 1]
```diff
@@ -41,9 +41,9 @@ export const createResponseItems = <T>(
 };
 
 export const createErrorResponse = (
-    httpError: HttpException, meta?: Meta, innerError?: InnerError, details?: ErrorObject[]): ErrorObject => {
+    httpError: HttpException, meta?: Meta, innerError?: InnerError, details?: ErrorObject[]): any => {
     if (httpError instanceof HttpException) {
```
Comment (CalvinHuynh) [inline src/helpers/response.helper.ts]: remove this, it is redundant

File: src/helpers/response.helper.ts [hunk 2]
```diff
@@ -53,6 +53,8 @@ export const createErrorResponse = (
                 details,
                 innerError
             },
-        };
+        },
+                                httpError.getStatus());
     }
+    throw new HttpException('Oops, something went wrong', HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (CalvinHuynh) [inline src/helpers/response.helper.ts]: same here

File: src/helpers/response.helper.ts (hunk 1 above)
Comment (RamonG92) [reply src/helpers/response.helper.ts]: Fixed

File: src/helpers/response.helper.ts (hunk 2 above)
Comment (RamonG92) [reply src/helpers/response.helper.ts]: Done here aswell

**Pull request merged**